traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -1179,7 +1180,7 @@ func selectMaxLastUpdatedQuery(where string) string {
 	select max(last_updated) as t from last_deleted l where l.table_name='deliveryservice') as res`
 }
 
-func getOldDetails(id int, tx *sql.Tx) (string, string, string, error) {
+func getOldDetails(id int, tx *sql.Tx) (string, string, string, error, int) {
```

Comment:
Previously, the assumption was that any error returned from this function would be 500-level, so returning a single error would be sufficient. Now that errors can be 400-level or 500-level, this function should return both a user-level error and a system-level error.